src/backend/distributed/citus--8.0-6--8.0-7.sql

```diff
@@ -0,0 +1,38 @@
+/* citus--8.0-6--8.0-7 */
+SET search_path = 'pg_catalog';
+
+CREATE VIEW citus.citus_lock_waits AS
+
+WITH citus_dist_stat_activity AS
+(
+  SELECT * FROM citus_dist_stat_activity
+),
+citus_dist_stat_activity_with_node_id AS
+(
+  SELECT 
+  citus_dist_stat_activity.*, (CASE citus_dist_stat_activity.master_query_host_name WHEN 'coordinator_host' THEN 0 ELSE pg_dist_node.nodeid END) as initiator_node_id
+  FROM
+  citus_dist_stat_activity LEFT JOIN pg_dist_node
+  ON
+  citus_dist_stat_activity.master_query_host_name = pg_dist_node.nodename AND 
+  citus_dist_stat_activity.master_query_host_port = pg_dist_node.nodeport
+)
+SELECT
```

Comment:
This query (and the query in the blog-post) generates duplicates when the commands are multi-shard. The main reason for that is `dump_local_wait_edges()` do not distinguish between master/worker queries and could have multiple entries for the same pair of distributed transactions.

We should probably have some kind of `DISTINCT ON` for `waiting.initiator_node_id, waiting.transaction_number,  blocking.initiator_node_idblocking.transaction_number`.

But, obviously, that's a vague idea that came to my mind right now. So, please ensure that `DISTINCT ON` works for us or not.

One another approach could be to add `worker_query` field to `dump_*_wait_edges()`, which requires some more extensive changes. So, let's first try to re-write the query such that we get rid of duplicates.